tt_eager/tt_dnn/op_library/rotary_embedding/kernels/dataflow/reader_rotary_embedding_llama_interleaved_start_id.cpp

```diff
@@ -0,0 +1,144 @@
+    constexpr uint32_t trans_mat_cb_id = get_compile_time_arg_val(3);
+    constexpr bool input_is_dram = get_compile_time_arg_val(4) == 1;
+    constexpr bool cos_is_dram = get_compile_time_arg_val(5) == 1;
+    constexpr bool sin_is_dram = get_compile_time_arg_val(6) == 1;
+    constexpr bool trans_mat_is_dram = get_compile_time_arg_val(7) == 1;
+    constexpr uint32_t Ht = get_compile_time_arg_val(8);
+    constexpr uint32_t Wt = get_compile_time_arg_val(9);
+    constexpr uint32_t HtWt = get_compile_time_arg_val(10);
+    constexpr uint32_t num_rows_per_core = get_compile_time_arg_val(11);
+    constexpr uint32_t num_sin_cos_rows_per_core = get_compile_time_arg_val(12);
+
+
+    constexpr uint32_t onetile = 1;
+    const uint32_t input_tile_bytes = get_tile_size(input_cb_id);
+    const DataFormat input_data_format = get_dataformat(input_cb_id);
+
+    const InterleavedAddrGenFast<input_is_dram> s0 = {
+        .bank_base_address = src_addr,
+        .page_size = input_tile_bytes,
+        .data_format = input_data_format
+    };
+
+    const uint32_t cos_tile_bytes = get_tile_size(cos_cb_id);
+    const DataFormat cos_data_format = get_dataformat(cos_cb_id);
+
+    const InterleavedAddrGenFast<cos_is_dram> s1 = {
+        .bank_base_address = cos_addr,
+        .page_size = cos_tile_bytes,
+        .data_format = cos_data_format
+    };
+
+    const uint32_t sin_tile_bytes = get_tile_size(sin_cb_id);
+    const DataFormat sin_data_format = get_dataformat(sin_cb_id);
+
+    const InterleavedAddrGenFast<sin_is_dram> s2 = {
+        .bank_base_address = sin_addr,
+        .page_size = sin_tile_bytes,
+        .data_format = sin_data_format
+    };
+
+    const uint32_t trans_mat_tile_bytes = get_tile_size(trans_mat_cb_id);
+    const DataFormat trans_mat_format = get_dataformat(trans_mat_cb_id);
+
+    const InterleavedAddrGenFast<trans_mat_is_dram> s3 = {
+        .bank_base_address = trans_mat_addr,
+        .page_size = trans_mat_tile_bytes,
+        .data_format = trans_mat_format
+    };
+
+    uint32_t input_curr_idx = start_row_idx * Wt;
+    uint32_t cos_sin_curr_idx = cos_sin_start_idx;
+    uint32_t trans_mat_curr_idx = 0;
+    uint32_t ht = start_row_idx;
+
+    // Read transformation matrix in CB (only once, because it will be reused)
+    cb_reserve_back(trans_mat_cb_id, onetile);
+    uint32_t trans_mat_l1_write_addr = get_write_ptr(trans_mat_cb_id);
+    noc_async_read_tile(trans_mat_curr_idx, s3, trans_mat_l1_write_addr);
+    noc_async_read_barrier();
+    cb_push_back(trans_mat_cb_id, onetile);
+
+    /*
+        Read a ublock of tiles from src to CB, and then push the ublock to unpacker
+
+        For example:
+            num_rows_per_core = 1 * 8 * 128 * 128 // 128 // 32 = 32
+            Ht = 4
+            Wt = 4
+    */
+    cb_reserve_back(sin_cb_id, Wt * num_sin_cos_rows_per_core);
+    cb_reserve_back(cos_cb_id, Wt * num_sin_cos_rows_per_core);
+    uint32_t sin_l1_write_addr = get_write_ptr(sin_cb_id);
```

Comment:
Not a big deal, but constants like `start_row_idx * Wt` can be runtime and avoided during the callback; constants like`Wt * num_sin_cos_rows_per_core` should be compile time.

Reply:
turned `start_row_idx * Wt` into a runtime arg and `Wt * num_sin_cos_rows_per_core` into a compile time arg. Made this change to both reader and compute kernels.